src/components/BulkEnrollmentModal/index.jsx

```diff
@@ -0,0 +1,219 @@
+      submitSucceeded,
+      onClose,
+      error,
+    } = this.props;
+    const {
+      mode,
+    } = this.state;
+
+    const errorMessageRef = this.errorMessageRef && this.errorMessageRef.current;
+
+    if (mode === 'remind' && submitSucceeded && submitSucceeded !== prevProps.submitSucceeded) {
+      onClose();
+    }
+
+    if (submitFailed && error !== prevProps.error && errorMessageRef) {
+      // When there is an new error, focus on the error message status alert
+      errorMessageRef.focus();
+    }
+  }
+
+  handleModalSubmit(formData) {
+    const {
+      enterpriseUuid,
+      sendBulkEnrollment,
+      selectedCourseRunKeys,
+    } = this.props;
+
+    const bulkEnrollmentPromises = selectedCourseRunKeys.map((selectedCourseRunKey) => {
+      const options = {
+        course_run_key: selectedCourseRunKey,
+        email: formData['bulk-enrollment-email-addresses'],
+        email_csv: formData['bulk-enrollment-email-csv'],
+      };
+      return sendBulkEnrollment(enterpriseUuid, options);
+    });
+
+    return Promise.all(bulkEnrollmentPromises)
+      .then((response) => {
+        this.props.onSuccess(response);
+      })
+      .catch((error) => {
+        throw new SubmissionError({
+          _error: [error.message],
+        });
+      });
+  }
+
+  resetModalWrapperState() {
+    this.setState({ open: false });
+  }
+
+  renderErrorMessage() {
+    const { error } = this.props;
+
+    return (
+      <div
+        ref={this.errorMessageRef}
+        tabIndex="-1"
+      >
+        <Alert
+          variant="danger"
+          dismissible
+        >
+          <Alert.Heading>Unable to assign codes</Alert.Heading>
+          <p className="alert-body">{error}</p>
+        </Alert>
+      </div>
+    );
+  }
+
+  renderFailedLearnersMessage(failedLearners) {
+    return (
+      <div
+        ref={this.errorMessageRef}
+        tabIndex="-1"
+      >
+        <Alert
+          variant="danger"
+          dismissible
+        >
+          <Alert.Heading>
+            Before you can invite learners to a course they must have a subscription license
+            </Alert.Heading>
+          <p className="alert-body">
+            The following emails addresses are not yet associated with a subscription.
+            Please take note of the following learners and invite them from the subscription management page.
+          </p>
+          {failedLearners.map((failedLearner) => <p className="alert-email mb-0">{failedLearner}</p>)}
+        </Alert>
+      </div>
+    );
+  }
+
+  renderBody() {
+    const {
+      failedLearners,
+      selectedCourseRunKeys,
+      submitFailed,
+    } = this.props;
+
+    return (
+      <>
+        {submitFailed && this.renderErrorMessage()}
+        {failedLearners.length > 0 && this.renderFailedLearnersMessage(failedLearners)}
+        <p className="courses-selected">Courses selected: {selectedCourseRunKeys.length}</p>
+        <form onSubmit={e => e.preventDefault()}>
```

Comment:
Can you use the Paragon Form components here? I think that'll make styling in line with what we want without thinking about it.
https://edx.github.io/paragon/components/forms
And likely
https://edx.github.io/paragon/components/validationformgroup

Reply:
Spent some time messing with this, doesn't look like there's an easy way to integrate the paragon forms with redux-forms (or at least I can't figure it out), going to punt on this and leave it consistent with the rest of the repo